src/components/KCopy/KCopy.vue

```diff
@@ -0,0 +1,274 @@
+          />
+        </span>
+      </KClipboardProvider>
+    </KTooltip>
+  </div>
+
+  <div
+    v-else
+    class="k-copy-badge"
+  >
+    <span
+      v-if="isBadge"
+      class="copy-badge-text"
+    >
+      {{ badgeText }}
+    </span>
+    <KBadge
+      v-if="isBadge"
+      appearance="custom"
+      background-color="#f9fafb"
+      border-color="#d7d8fe"
+      class="id-badge"
+      color="#473cfb"
+      is-bordered
+      shape="rectangular"
+    >
+      <KCopy
+        :badge-text="badgeText"
+        class="id-badge-content"
+        :content="content"
+        :content-tooltip="contentTooltip"
+        :copy-tooltip="copyTooltip"
+        :monospace="monospace"
+        :success-tooltip="successTooltip"
+        :truncate="truncate"
+        :truncation-limit="truncationLimit"
+      />
+    </KBadge>
+  </div>
+</template>
+
+<script setup lang="ts">
+import type { PropType } from 'vue'
+import { computed, ref } from 'vue'
+import KBadge from '@/components/KBadge/KBadge.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KClipboardProvider from '@/components/KClipboardProvider'
+import KTooltip from '@/components/KTooltip/KTooltip.vue'
+
+const props = defineProps({
+  // text displayed before the copyable content when `isBadge` is true
+  badgeText: {
+    type: String,
+    default: '',
+  },
+  // the copyable text
+  content: {
+    type: String,
+    required: true,
+  },
+  // tooltip text displayed on hover over the `content`
+  contentTooltip: {
+    type: String,
+    default: '',
+  },
+  // tooltip text displayed on hover over copy button
+  copyTooltip: {
+    type: String,
+    default: 'Copy',
+  },
+  // formatting for copyable content (default, hidden, redacted, deleted)
+  format: {
+    type: String as PropType<'default' | 'hidden' | 'redacted' | 'deleted'>,
+    required: false,
+    default: 'default',
+    validator: (val: string) => ['default', 'hidden', 'redacted', 'deleted'].includes(val),
+  },
+  // whether or not to display as a badge
+  isBadge: {
+    type: Boolean,
+    default: false,
+  },
+  // false if `isBadge` is true
+  monospace: {
+    type: Boolean,
+    default: true,
+  },
+  // whether or not the content should be truncated
+  truncate: {
+    type: Boolean,
+    default: false,
+  },
+  // tooltip text displayed on successful copy
+  successTooltip: {
+    type: String,
+    default: 'Copied',
+  },
+  // number of characters to truncate at
+  truncationLimit: {
+    type: Number,
+    default: 8,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'copied', val: string): void
+}>()
+
+const hasSuccessTooltip = computed((): boolean => !!(props.copyTooltip && props.successTooltip))
+const tooltipText = ref(props.copyTooltip)
+
+const iconTitle = computed(() => {
+  if (props.badgeText) {
+    return `Copy ${props.badgeText}`
+  }
+  return 'Copy'
+})
+
+const widthStyle = computed(() => {
+  return {
+    maxWidth: props.truncate ? props.truncationLimit + 'ch' : undefined,
```

Comment:
Can we use `px` instead of `ch` unit here? We don't want to be using relative units